### PR TITLE
Add missing distribute() calls

### PR DESCRIPTION
**Pull request by class4kayaker**
Fix missing calls to constraints.distribute() mentioned in #757.

I would appreciate a second set of eyes to confirm whether any additional explanatory text is needed, or if this problem had already been fixed.

The steps noted in 
- 31: changed
- 32: exists
- 33: no constraints to apply
- 42: exists
- 43: changed

### CODE REVIEW

**Comment by masterleinad** [comment]
Can you confirm that the results of step-31 and step-43 stay the same? If they differ noticeably, the results section needs to be updated as well. 
I can imagine that the pictures can remain the same, but the output printed below "Results in 2d" in step-31 might be slightly different.

File: examples/step-31/step-31.cc [hunk 1]
```diff
@@ -2081,8 +2081,13 @@ namespace Step31
     temperature_solution = tmp[0];
     old_temperature_solution = tmp[1];
 
+    temperature_constraints.distribute(temperature_solution);
```
Comment (masterleinad) [inline examples/step-31/step-31.cc]: We comment on what we are doing in this function quite detailed. Could you put a comment in the comment above saying that we are also enforcing constraints to make the interpolated solution conforming on the new mesh?

**Comment by class4kayaker** [comment]
The results for step-31 are not identical to those in the results document. The alteration has a smaller number of CG iterations and a smaller temperature range than that listed in the results section, though the number of cells for the first few steps does match. Visual examination of the results is similar to that in the given images.

A comparison to a test on my local machine has the number of CG and GMRES iterations be identical past the 50th timestep with some variations in temperature range and current time.

A paraview generated image for the results of the change for the first timestep
soln-0-new
soln-0-master

For step-43, the stdout output does not significantly diverge until timestep 123, with only a few earlier variations in GMRES iterations or last digit of time.

Paraview generated image for iteration 200:
soln-200
soln-200-master

As such, I believe the results are reasonable though a second opinion would be appreciated.

File: examples/step-31/step-31.cc (hunk 1 above)
Comment (class4kayaker) [inline examples/step-31/step-31.cc]: The surrounding comments have been slightly adjusted. I would appreciate any suggestions you might have.

**Comment by masterleinad** [comment]
Could you update the output for "Results in 3d" in step-31 as well?
For the pictures just leave a note that they were created with a version where we didn't enforce the constraints after transferring the solution.

Apart from that this looks good to me. Thanks for looking into this issue!

**Pull request merged**